packages/accounts/src/msca/e2e-tests/multisig-modular-account.e2e.test.ts

```diff
@@ -0,0 +1,572 @@
+    };
+
+    const signature1 = await provider1.account.signTypedDataWith6492({
+      types,
+      primaryType,
+      message,
+    });
+
+    // console.log("signature1: ", signature1);
+
+    const signature2 = await provider2.account.signTypedDataWith6492({
+      types,
+      primaryType,
+      message,
+    });
+
+    // console.log("signature2: ", signature2);
+
+    const combined = formatSignatures([
+      {
+        userOpSigType: "ACTUAL",
+        signerType: "EOA",
+        signature: signature1,
+        signer: await signer1.getAddress(),
+      },
+      {
+        userOpSigType: "ACTUAL",
+        signerType: "EOA",
+        signature: signature2,
+        signer: await signer2.getAddress(),
+      },
+    ]);
+
+    // console.log("combined: ", combined);
+
+    expect(
+      await provider1.verifyTypedData({
+        address: provider1.getAddress(),
+        types,
+        primaryType,
+        message,
+        signature: combined,
+      })
+    ).toBe(true);
+  });
+
+  it("should correctly sign and verify 6492 signatures over messages for an undeployed account", async () => {
+    // Add a fourth signer to change the counterfactual (not yet deployed)
+
+    const signer4: SmartAccountSigner<HDAccount> =
+      LocalAccountSigner.mnemonicToAccountSigner(OWNER_MNEMONIC, {
+        accountIndex: 3,
+      });
+
+    const provider1 = await givenConnectedProvider({
+      signer: signer1,
+      chain,
+      owners: [...owners, await signer4.getAddress()],
+      threshold,
+    });
+    const provider2 = await givenConnectedProvider({
+      signer: signer2,
+      chain,
+      owners: [...owners, await signer4.getAddress()],
+      threshold,
+    });
+
+    const {
+      account: { address },
+    } = provider1;
+    expect(address).toEqual("0xB77423329491BAF4b7B904887627C55Cd53968f8");
+
+    const message = "test";
+
+    const signature1 = await provider1.account.signMessage({ message });
+
+    // console.log("signature1: ", signature1);
+
+    const signature2 = await provider2.account.signMessage({ message });
+
+    // console.log("signature2: ", signature2);
+
+    const combined = formatSignatures([
+      {
+        userOpSigType: "ACTUAL",
+        signerType: "EOA",
+        signature: signature1,
+        signer: await signer1.getAddress(),
+      },
+      {
+        userOpSigType: "ACTUAL",
+        signerType: "EOA",
+        signature: signature2,
+        signer: await signer2.getAddress(),
+      },
+    ]);
+
+    // console.log("combined: ", combined);
+
+    const [, factoryCalldata] = parseFactoryAddressFromAccountInitCode(
+      await provider1.account.getInitCode()
+    );
+
+    const wrappedSig = wrapSignatureWith6492({
+      factoryAddress: provider1.account.getFactoryAddress(),
+      factoryCalldata,
+      signature: combined,
+    });
+
+    // todo: can we override verifyMessage & verifyTypedData to internally only use the public client, to prevent the EIP-684 issue?
```

Comment:
in most cases I dunno if people will run into this because they won't be verifying with the same client they just signed with, but agree it's an annoying edge case